specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,264 @@
+
+CHECK_FAILURE(frame->add_LaunchingRegisteredProtocol(
+    Callback<ICoreWebView2FrameLaunchingRegisteredProtocolEventHandler>(
+        [this](
+            ICoreWebView2Frame* sender,
+            ICoreWebView2LaunchingRegisteredProtocolEventArgs* args) {
+                wil::unique_cotaskmem_string uri;
+                CHECK_FAILURE(args->get_Uri(&uri));
+                if (wcscmp(uri.get(), L"calculator://") == 0)
+                {
+                    // If this matches our desired protocol, then suppress the
+                    // popup dialog.
+                    CHECK_FAILURE(args->put_Handled(TRUE));
+                }
+                else
+                {
+                    // Otherwise revoke permissions previously granted to this protocol 
+                    // from this origin. This will trigger the dialog to appear to allow the user
+                    // to confirm or deny the protocol launch.
+                    CHECK_FAILURE(args->put_RevokeProtocolPermissionsPerOrigin(TRUE));
+                }
+                return S_OK;
+        })
+        .Get(),
+    &m_frameLaunchingRegisteredProtocolToken));
+    
+``` 
+
+## .NET and WinRT
+
+```c #
+
+void WebView_LaunchingRegisteredProtocol(object target, CoreWebView2LaunchingRegisteredProtocolEventArgs e)
+{
+    if (e.Uri == "calculator:///") 
+    {
+        // If this matches our desired protocol, then suppress the popup dialog.
+        e.Handled = true;
+    }
+    else 
+    {
+        // Otherwise allow the popup dialog, and allow the user to decide 
+        // whether or not to allow the protocol to launch.
+        e.Handled = false;
+    }
+    webView.CoreWebView2.FrameCreated += (sender, args) =>
+    {
+        // Apply the same logic as above to non-main frame raising the event.
+        args.Frame.LaunchingRegisteredProtocol += (frameSender, LaunchingRegisteredProtocolArgs) =>
+        {
+            if (LaunchingRegisteredProtocolArgs.Uri == "calculator:///") 
+            {
+                // If this matches our desired protocol, then suppress the popup dialog.
+                e.Handled = true;
+            }
+            else 
+            {
+                // Otherwise revoke permissions previously granted to this protocol 
+                // from this origin. This will trigger the dialog to appear to allow the user
+                // to confirm or deny the protocol launch.
+                e.RevokeProtocolPermissionsPerOrigin = true;
+            }
+        };
+    };
+}
+
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+    
+```IDL
+// This is the ICoreWebView2_11 interface.
+[uuid(cc39bea3-d6d8-471b-919f-da253e2fbf03), object, pointer_default(unique)] 
+interface ICoreWebView2_11 : IUnknown {
+  /// Add an event handler for the `RegisteredProtocol` event.
+  /// The RegisteredProtocol event fires when a launch request is made to a protocol
```

Comment:
We should define a launch request. Is that when the WebView2 is trying to navigate to a protocol that is registered with the OS? Are there other ways for this to happen besides a navigation?

Reply:
This only occurs when trying to launch a url through a navigation.